### ISSUE TITLE
Any good examples where this tool is used?

### ISSUE THREAD

**Issue by ClassyCircuit** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#437]
Hi,
Are there any open-source projects where this tool is used? I want to see how others are interacting with it, so perhaps, I could use some of the best practices in my project.

Thanks!

**Comment by hero3616** [comment]
I use this tool in many projects (not open source) where table number is less than ~200.

**Comment by ClassyCircuit** [comment]
@hero3616 which design patterns are you using in conjunction with this tool? 
I implemented repository pattern, but I am having trouble with code duplication, not sure what's the best way to improve it.

**Comment by sjh37** [comment]
I tend to split this entity generation into model/domain folder which just contains the pocos.that project is used by the application layer. Then I have a persistence layer containing the generated db contect. The application layer has the db context interfaces that the persistence layer implements. I show how to break up the generation in the pluralsight course. Either break it up into separate folders or separate projects.

Hard for people to demo this unless they use a free database such as adventure works or northwind. Perhaps I should create this to show a proper 4 layer DDD architecture.

**Comment by ClassyCircuit** [comment]
@sjh37 yes that would be very useful! I have also watched your tutorials on pluralsight - they are good for understanding how to generate stuff and how to use your tool, but I am afraid, I did not understand what would be a good way of layering the whole architecture.

**Comment by ClassyCircuit** [comment]
@sjh37 Could you share a project where you've implemented DDD architecture together with Reverse POCO generator? It would be of much help! Thanks

**Comment by sjh37** [comment]
Answered above. The wiki now covers splitting entities, context and configuration across projects with `Settings.ElementsToGenerate` and one `.tt` per project. Closing.
